contracts/Delay.sol

```diff
@@ -90,13 +131,49 @@ contract Delay is AragonApp, IForwarder {
         emit ExecutedScript(_delayedScriptId);
     }
 
-    function canExecute(uint256 _scriptId) public returns (bool) {
+    function canPause(uint256 _scriptId) public view scriptExists(_scriptId) returns (bool) {
+        return !_isExecutionPaused(_scriptId);
+    }
+
+    function canResume(uint256 _scriptId) public view scriptExists(_scriptId) returns (bool) {
+        return _isExecutionPaused(_scriptId);
+    }
+
+    /**
+    * @notice Tells whether a script with ID #`_scriptId` can be executed or not
+    * @param _scriptId The ID of the script to execute
+    */
+    function canExecute(uint256 _scriptId) public scriptExists(_scriptId) returns (bool) {
+
+        if (_isExecutionPaused(_scriptId))
+            return false;
+
+        bool withinExecutionWindow = now > delayedScripts[_scriptId].executionTime;
+        return withinExecutionWindow;
+    }
+
+    function _isExecutionPaused(uint256 _scriptId) internal view returns (bool) {
+        return delayedScripts[_scriptId].pausedAt != 0;
+    }
+
+    function _pauseExecution(uint256 _scriptId) internal {
+        delayedScripts[_scriptId].pausedAt = now;
+        emit ExecutionPaused(_scriptId);
+    }
+
+    function _resumeExecution(uint256 _scriptId) internal {
         DelayedScript storage delayedScript = delayedScripts[_scriptId];
 
-        bool delayedScriptExists = delayedScript.timeSubmitted != 0;
-        bool withinExecutionWindow = now > delayedScript.timeSubmitted + executionDelay;
+        delayedScript.executionTime = delayedScript.executionTime.add(now.sub(delayedScript.pausedAt));
```

Comment:
I would make this clearer by putting `now.sub(delayedScript.pausedAt)` in a `timePaused` or similar var.